src/views/SocialView.vue

```diff
@@ -1,111 +1,293 @@
+        node10: [],
+        node7: []
+      },
+
+      sessionAuthor: null, // Get from store
+      basePath: String, // none
+      loading: true
+    }
   },
   methods: {
-    getAuthors() {
-      axios.get(
-        '/authors/')
-      .then((res) => {
-        this.authors = res.data
-        console.log(res.data)
-      })
-      .catch(function(err) {
-        console.log(err);
-      });
+    // Used to transform "followers" -> "follower"
+    textTransform (word, singularize, capitalize) {
+      try {
+        if (capitalize) {
+          word = word?.charAt(0).toUpperCase() + word?.slice(1)
+        }
+        if (singularize) {
+          word = word?.replace(/s$/, '')
+        }
+        return word
+      } catch {
+        console.log('Failed on: ' + word)
+        console.table(word)
+      }
+    },
+
+    // Related to the friend request logic
+    handleRequest (state, existingFollow) {
+      // Create a new follow object with a state (accepted/rejected)
+      const followToSend = followTemplate
+
+      followToSend.state = state
+      followToSend.actor = existingFollow.actor
+      followToSend.object = this.sessionAuthor
+
+      console.table(followToSend)
+
+      this.$localNode
+        .post(`${this.basePath}/inbox/`, followToSend)
+        .then(() => {
+          alert(`${state} the Request`) // Computed pendingRequests should auto-update
+          if (state === 'Accepted') {
+            this.inbox.push(followToSend) // Update local version of allRequests
+            this.moveToFriendsOrFollower(followToSend.actor)
+          }
+        })
+        .catch((err) => {
+          console.log(err)
+        })
+    },
+
+    moveToFriendsOrFollower (acceptedFollow) {
+      // Helper function to move people from the requests to the appropriate field for the local
+
+      // Sheesh this funciton is incredibly inefficient but that's web dev for you am I right?
+      const allRequests = this.inbox.filter(item => item.type === 'Follow')
+
+      const recipricalFollow = acceptedFollow
+      recipricalFollow.author = acceptedFollow.actor
+      recipricalFollow.actor = acceptedFollow.author
+
+      if (allRequests.includes(acceptedFollow) && allRequests.includes(recipricalFollow)) {
+        this.friendliesProxy.friends.push(acceptedFollow.actor)
+      } else {
+        this.friendliesProxy.followers.push(acceptedFollow.actor)
+      }
+    },
+
+    sendFollow (author) {
+      // Create follow object
+      const followToSend = followTemplate
+      followToSend.actor = this.sessionAuthor
+      followToSend.object = author
+
+      if (author.id === this.sessionAuthor.id) {
+        alert('You can\'t send a follow request to yourself')
+        return
+      }
+
+      // Set up axios
+      const hostNode = getAxiosTarget(author.host) // Gets the corresponding axios instance
+      const endpoint = new URL(author.id).pathname.replace(/^\/api/, '')
+
+      hostNode.post(`${endpoint}/inbox/`, followToSend)
```

Comment:
For the follow request:
```
const followTemplate = {
  type: 'Follow',
  actor: Object,
  object: Object,
  state: null // Null if not accepted/rejected
}
```
Could we also send it to the endpoint /authors/{AUTHOR_ID}/remote-requests/ by doing a PUT request whenever **local to remote** friend requests are sent?

This is the other endpoint (besides retrieving the authorization tokens of the remote groups) that is currently not working due to database issues, but will help to determine followees and friends, since it helps me to keep track of the **local to remote** friend requests sent. 

When local author A sends a friend request to remote author B, it gets sent to remote author B's inbox, and then having this same **local to remote** friend request sent to  /authors/{AUTHOR_ID}/remote-requests/, where AUTHOR_ID is the author who sent a **local to remote** friend request, in which case, that would be local author A. This helps to check if remote author B has accepted local author A's friend request **yet** (i.e. by checking if local author A has appeared in the followers of remote author B yet). It doesn't matter if a state is included or not.

Example (source: a previous commit from `deploy-frontend` branch): 
See https://github.com/CMPUT404W23T05/Social-Distribution-Front-End/commit/45ff24fce01e69cebbf94c88e544881bd81a5593

Reply:
You can indeed replace post with put if that's the behaviour you want. Should work just fine with the front-end.